### ISSUE TITLE
ConnectAsyncV6IPEndPointToDualHost_Success failed in CI on Linux

### ISSUE THREAD

**Issue by stephentoub** [dotnet/corefx#3748]
http://dotnet-ci.cloudapp.net/job/dotnet_corefx_linux_debug_tst_prtest/3215/console
```
10:50:47    System.Net.Sockets.Tests.DualMode.ConnectAsyncV6IPEndPointToDualHost_Success [FAIL]
10:50:47       System.Net.Sockets.SocketException : Unknown error 10054
10:50:47       Stack Trace:
10:50:47             at System.Net.Sockets.Tests.DualMode.DualModeConnectAsync_IPEndPointToHost_Helper(IPAddress connectTo, IPAddress listenOn, Boolean dualModeServer)
10:50:47             at System.Net.Sockets.Tests.DualMode.ConnectAsyncV6IPEndPointToDualHost_Success()
10:50:47             at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
10:50:47             at System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments)
10:50:47             at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
```

**Comment by stephentoub** [comment]
Failed again here:
http://dotnet-ci.cloudapp.net/job/dotnet_corefx_linux_debug_tst_prtest/3991/console
```
08:54:20 Starting:    System.Linq.Parallel.Tests
08:54:20    System.Net.Sockets.Tests.DualMode.ConnectAsyncV6IPEndPointToDualHost_Success [FAIL]
08:54:20       System.Net.Sockets.SocketException : Unknown error 10054
08:54:20       Stack Trace:
08:54:21 Finished:    System.Reflection.TypeExtensions.Tests
08:54:21 
08:54:21 === TEST EXECUTION SUMMARY ===
08:54:21    System.Reflection.TypeExtensions.Tests  Total: 599, Errors: 0, Failed: 0, Skipped: 0, Time: 1.690s
08:54:21             at System.Net.Sockets.Tests.DualMode.DualModeConnectAsync_IPEndPointToHost_Helper(IPAddress connectTo, IPAddress listenOn, Boolean dualModeServer)
08:54:21             at System.Net.Sockets.Tests.DualMode.ConnectAsyncV6IPEndPointToDualHost_Success()
```

**Comment by stephentoub** [comment]
Related failure:
http://dotnet-ci.cloudapp.net/job/dotnet_corefx_linux_debug_tst_prtest/4166/console 
```
17:34:18    System.Net.Sockets.Tests.DualMode.ConnectAsyncV6IPEndPointToV6Host_Success [FAIL]
17:34:18       System.Net.Sockets.SocketException : Unknown error 10054
17:34:18       Stack Trace:
17:34:18 Finished:    System.Composition.Tests
17:34:18 
17:34:18 === TEST EXECUTION SUMMARY ===
17:34:18    System.Composition.Tests  Total: 142, Errors: 0, Failed: 0, Skipped: 0, Time: 4.834s
17:34:18             at System.Net.Sockets.Tests.DualMode.DualModeConnectAsync_IPEndPointToHost_Helper(IPAddress connectTo, IPAddress listenOn, Boolean dualModeServer)
17:34:18             at System.Net.Sockets.Tests.DualMode.ConnectAsyncV6IPEndPointToV6Host_Success()
```

**Comment by stephentoub** [comment]
Another failure here:
http://dotnet-ci.cloudapp.net/job/dotnet_corefx_linux_debug_tst_prtest/4236/console
```
13:36:44    System.Net.Sockets.Tests.DualMode.ConnectAsyncV6IPEndPointToV6Host_Success [FAIL]
13:36:44       System.Net.Sockets.SocketException : Unknown error 10054
13:36:44       Stack Trace:
13:36:45             at System.Net.Sockets.Tests.DualMode.DualModeConnectAsync_IPEndPointToHost_Helper(IPAddress connectTo, IPAddress listenOn, Boolean dualModeServer)
13:36:45             at System.Net.Sockets.Tests.DualMode.ConnectAsyncV6IPEndPointToV6Host_Success()
```

**Comment by justinvp** [comment]
Failed here:

http://dotnet-ci.cloudapp.net/job/dotnet_corefx_linux_debug_tst_prtest/4581/console

```
18:02:00    System.Net.Sockets.Tests.DualMode.ConnectAsyncV6IPEndPointToDualHost_Success [FAIL]
18:02:00       System.Net.Sockets.SocketException : Unknown error 10054
18:02:00       Stack Trace:
18:02:01             at System.Net.Sockets.Tests.DualMode.DualModeConnectAsync_IPEndPointToHost_Helper(IPAddress connectTo, IPAddress listenOn, Boolean dualModeServer)
18:02:01             at System.Net.Sockets.Tests.DualMode.ConnectAsyncV6IPEndPointToDualHost_Success()
```

**Comment by ericeil** [comment]
Looks like the same underlying cause as #3494.

**Comment by ericeil** [comment]
Closed with #5907
